ion-schema/src/ion_path.rs

```diff
@@ -2,64 +2,80 @@ use ion_rs::value::owned::Element;
 use ion_rs::value::Builder;
 use ion_rs::Symbol;
 use std::fmt;
-use std::fmt::Formatter;
+use std::fmt::{Debug, Formatter};
 
-#[derive(Clone, PartialEq)]
+/// Represents a single element in Ion Path which is either an index value or a field name depending on its parent container type
+#[derive(Clone, PartialEq, PartialOrd)]
 pub enum IonPathElement {
-    IndexedElement { index: usize },
-    Field { name: String },
+    Index(usize),
+    Field(String),
 }
 
 // TODO: This can be removed once we have a complete Display for violation
 // This is currently used by schema sandbox to print nested violations
 impl fmt::Debug for IonPathElement {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         match &self {
-            IonPathElement::IndexedElement { index } => {
-                write!(f, "{index}")
+            IonPathElement::Index(index) => {
+                write!(f, "{}", Element::from(*index as i64))
             }
-            IonPathElement::Field { name } => {
-                write!(f, ".{name}")
+            IonPathElement::Field(name) => {
+                write!(f, "{}", Element::from(Symbol::from(name.as_str())))
             }
         }
     }
 }
 
 impl fmt::Display for IonPathElement {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        match &self {
-            IonPathElement::IndexedElement { index } => {
-                write!(f, "[{index}]")
-            }
-            IonPathElement::Field { name } => {
-                write!(f, ".{name}")
-            }
-        }
+        Debug::fmt(&self, f)
     }
 }
 
-#[derive(Default, Clone, PartialEq)]
+/// Represents an IonPath associated with given Ion value for which the violation occurred.
+/// IonPath consists of IonPathElements where each element could either be an index value or a field name.
+///
+/// Note: IonPath is Displayed as an SExpression where each element of SExpression is either an index value or a field name.
+///
+/// Example of IonPath for an Ion value is as following:
+/// ```ion
+/// {
+///     greetings: [ "hello", "hi", "hey" ]
+/// }
+/// ```
+///
+/// For an Ion value given as above if the violation occurred for Ion value "hi" then the associated
```

Comment:
I suggest starting with an example of the empty path (`()`) referring to the top level struct, then making subsequent examples show increasingly nested paths.

Reply:
Got it. I will add that.